src/bin/server/transcoding.rs

```diff
@@ -0,0 +1,120 @@
+use actix_web;
+use actix_web::dev::Handler;
+use actix_web::error;
+use actix_web::http;
+use actix_web::App;
+use actix_web::AsyncResponder;
+use actix_web::FromRequest;
+use actix_web::HttpRequest;
+use actix_web::Path;
+use actix_web::State;
+
+use forte_core::models::Song;
+
+use server::graphql::AppState;
+use server::stream::RangeStream;
+use server::transcoder::{TranscodeMessage, TranscodeTarget};
+
+use diesel;
+
+use uuid::Uuid;
+
+use futures::future;
+use futures::Future;
+
+use lru_disk_cache::ReadSeek;
+
+use std::io;
+use std::io::SeekFrom;
+use std::path::PathBuf;
+
+pub struct TranscodedSongHandler {
+    transcode_target: TranscodeTarget,
+}
+
+impl TranscodedSongHandler {
+    pub fn new(target: TranscodeTarget) -> TranscodedSongHandler {
+        TranscodedSongHandler {
+            transcode_target: target,
+        }
+    }
+}
+
+impl Handler<AppState> for TranscodedSongHandler {
+    type Result = actix_web::FutureResponse<RangeStream<Box<ReadSeek>>>;
+
+    fn handle(&mut self, req: HttpRequest<AppState>) -> Self::Result {
+        let state: State<AppState> = State::from_request(&req, &());
+        let song_id_path: Path<Uuid> =
+            Path::from_request(&req, &()).expect("song id path parameter missing");
```

Comment:
Return an error instead of panicking.

Reply:
Decided to panic because this should only happen in the case of a programmer error. It will never happen when using `TranscodedHandlerAppExt`.

Also, returning an error is hard as the return type is a Future.